tvx/suite_messages_paych.go

```diff
@@ -0,0 +1,229 @@
+		}, chain.Nonce(1), chain.Value(big_spec.Zero()))
+		v.ApplyMessages = append(v.ApplyMessages, Message{Bytes: chain.MustSerialize(msg)})
+		td.ApplyOk(msg)
+
+		var pcState paych_spec.State
+		td.GetActorState(paychAddr, &pcState)
+		assert.Equal(t, 1, len(pcState.LaneStates))
+		ls := pcState.LaneStates[0]
+		assert.Equal(t, pcAmount, ls.Redeemed)
+		assert.Equal(t, pcNonce, ls.Nonce)
+		assert.Equal(t, pcLane, ls.ID)
+
+		postroot := td.GetStateRoot()
+
+		v.CAR = td.MustMarshalGzippedCAR(preroot, postroot)
+		v.Pre.StateTree.RootCID = preroot
+		v.Post.StateTree.RootCID = postroot
+
+		// encode and output
+		enc := json.NewEncoder(os.Stdout)
+		if err := enc.Encode(&v); err != nil {
+			return err
+		}
+
+		return nil
+	}("happy path update")
+	if err != nil {
+		return err
+	}
+
+	err = func(testname string) error {
+		td := drivers.NewTestDriver()
+
+		v := newEmptyMessageVector()
+
+
+		// create the payment channel
+		sender, _ := td.NewAccountActor(drivers.SECP, initialBal)
+		receiver, receiverID := td.NewAccountActor(drivers.SECP, initialBal)
+		paychAddr := chain.MustNewIDAddr(chain.MustIdFromAddress(receiverID) + 1)
+		initRet := td.ComputeInitActorExecReturn(sender, 0, 0, paychAddr)
+
+		preroot := td.GetStateRoot()
+
+		msg := td.MessageProducer.CreatePaymentChannelActor(sender, receiver, chain.Value(toSend), chain.Nonce(0))
+		v.ApplyMessages = append(v.ApplyMessages, Message{Bytes: chain.MustSerialize(msg)})
+		td.ApplyExpect(
+			msg,
+			chain.MustSerialize(&initRet))
+		td.AssertBalance(paychAddr, toSend)
+
+		msg = td.MessageProducer.PaychUpdateChannelState(sender, paychAddr, &paych_spec.UpdateChannelStateParams{
+			Sv: paych_spec.SignedVoucher{
+				ChannelAddr:     paychAddr,
+				TimeLockMin:     abi_spec.ChainEpoch(0),
+				TimeLockMax:     0, // TimeLockMax set to 0 means no timeout
+				SecretPreimage:  nil,
+				Extra:           nil,
+				Lane:            1,
+				Nonce:           1,
+				Amount:          toSend, // the amount that can be redeemed by receiver,
+				MinSettleHeight: 0,
+				Merges:          nil,
+				Signature: &crypto_spec.Signature{
+					Type: crypto_spec.SigTypeBLS,
+					Data: []byte("signature goes here"),
+				},
+			},
+		}, chain.Nonce(1), chain.Value(big_spec.Zero()))
+
+		v.ApplyMessages = append(v.ApplyMessages, Message{Bytes: chain.MustSerialize(msg)})
+
+		td.ApplyOk(msg)
+
+		// settle the payment channel so it may be collected
+
+		msg = td.MessageProducer.PaychSettle(receiver, paychAddr, nil, chain.Value(big_spec.Zero()), chain.Nonce(0))
+		v.ApplyMessages = append(v.ApplyMessages, Message{Bytes: chain.MustSerialize(msg)})
+		settleResult := td.ApplyOk(msg)
+
+		// advance the epoch so the funds may be redeemed.
+		td.ExeCtx.Epoch += paych_spec.SettleDelay
+
+		msg = td.MessageProducer.PaychCollect(receiver, paychAddr, nil, chain.Nonce(1), chain.Value(big_spec.Zero()))
+		v.ApplyMessages = append(v.ApplyMessages, Message{Epoch: &td.ExeCtx.Epoch, Bytes: chain.MustSerialize(msg)})
```

Comment:
The `collect` message should be set `paych_spec.SettleDelay` epochs later than the `settle` I think.